hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/PendingDeletionEndpoint.java

```diff
@@ -0,0 +1,115 @@
+
+/**
+ * REST API endpoint that provides metrics and information related to
+ * pending deletions. It responds to requests on the "/pendingDeletion" path
+ * and produces application/json responses.
+ */
+@Path("/pendingDeletion")
+@Produces("application/json")
+@AdminOnly
+public class PendingDeletionEndpoint {
+  private static final Logger LOG = LoggerFactory.getLogger(PendingDeletionEndpoint.class);
+  private final ReconGlobalMetricsService reconGlobalMetricsService;
+  private final DataNodeMetricsService dataNodeMetricsService;
+  private final StorageContainerLocationProtocol scmClient;
+
+  @Inject
+  public PendingDeletionEndpoint(ReconGlobalMetricsService reconGlobalMetricsService,
+      DataNodeMetricsService dataNodeMetricsService,
+      StorageContainerLocationProtocol scmClient) {
+    this.reconGlobalMetricsService = reconGlobalMetricsService;
+    this.dataNodeMetricsService = dataNodeMetricsService;
+    this.scmClient = scmClient;
+  }
+
+  @GET
+  public Response getPendingDeletionByComponent(@QueryParam("component") String component) {
+    if (component == null || component.isEmpty()) {
+      return Response.status(Response.Status.BAD_REQUEST)
+          .entity("component query parameter is required").build();
+    }
+    final String normalizedComponent = component.trim().toLowerCase();
+    switch (normalizedComponent) {
+    case "dn":
+      return handleDataNodeMetrics();
+    case "scm":
+      return handleScmPendingDeletion();
+    case "om":
+      return handleOmPendingDeletion();
+    default:
+      return Response.status(Response.Status.BAD_REQUEST)
+          .entity("component query parameter must be one of dn, scm, om").build();
+    }
+  }
+
+  private Response handleDataNodeMetrics() {
+    DataNodeMetricsService.MetricCollectionStatus status = dataNodeMetricsService.getTaskStatus();
+    if (status != DataNodeMetricsService.MetricCollectionStatus.IN_PROGRESS) {
+      CompletableFuture.runAsync(dataNodeMetricsService::startTask);
+    }
+    DataNodeMetricsServiceResponse response = dataNodeMetricsService.getCollectedMetrics();
+    if (response.getStatus() == DataNodeMetricsService.MetricCollectionStatus.SUCCEEDED) {
+      return Response.ok(response).build();
+    } else {
+      return Response.accepted(response).build();
+    }
+  }
+
+  private Response handleScmPendingDeletion() {
+    try {
+      HddsProtos.DeletedBlocksTransactionSummary summary = scmClient.getDeletedBlockSummary();
+      if (summary == null) {
+        return Response.noContent()
+            .build();
+      }
+      ScmPendingDeletion pendingDeletion = new ScmPendingDeletion(
+          summary.getTotalBlockSize(),
+          summary.getTotalBlockReplicatedSize(),
+          summary.getTotalBlockCount());
+      return Response.ok(pendingDeletion).build();
+    } catch (Exception e) {
+      LOG.error("Error getting SCM pending deletion", e);
+      return Response.status(Response.Status.INTERNAL_SERVER_ERROR)
```

Comment:
The error handling of OM, SCM and DN request failure is not consistent, 
a. SCM, INTERNAL_SERVER_ERROR
b. OM, key 0, dir INTERNAL_SERVER_ERROR
c. DN, -1